code/core/src/phone/java/com/adobe/marketing/mobile/services/DeviceInfoService.java

```diff
@@ -214,71 +214,15 @@ public String getMobileCarrierName() {
-
-                if (activeNetworkInfo != null) {
-                    // At this point we have everything that we need to accurately determine the
-                    // current connectivity status
-                    return activeNetworkInfo.isAvailable() && activeNetworkInfo.isConnected()
-                            ? DeviceInfoService.ConnectionStatus.CONNECTED
-                            : DeviceInfoService.ConnectionStatus.DISCONNECTED;
-                } else {
-                    // Per Android documentation getActiveNetworkInfo() will return null if no
-                    // default network is currently active.
-                    // If no default network is currently active we can assume that we don't have
-                    // connectivity.
-                    Log.debug(
-                            ServiceConstants.LOG_TAG,
-                            LOG_TAG,
-                            "Unable to determine connectivity status due to there being no default"
-                                    + " network currently active");
-                }
-            } else {
-                Log.debug(
-                        ServiceConstants.LOG_TAG,
-                        LOG_TAG,
-                        "Unable to determine connectivity status due to the system service"
-                                + " requested being unrecognized");
-            }
-        } catch (NullPointerException e) {
-            Log.debug(
-                    ServiceConstants.LOG_TAG,
-                    LOG_TAG,
-                    String.format(
-                            "Unable to determine connectivity status due to an unexpected error"
-                                    + " (%s)",
-                            e.getLocalizedMessage()));
-        } catch (SecurityException e) {
-            Log.debug(
-                    ServiceConstants.LOG_TAG,
-                    LOG_TAG,
-                    String.format(
-                            "Unable to access connectivity status due to a security error (%s)",
-                            e.getLocalizedMessage()));
-        } catch (Exception e) {
-            Log.debug(
-                    ServiceConstants.LOG_TAG,
-                    LOG_TAG,
-                    String.format(
-                            "Unable to access connectivity status due to an unexpected error (%s)",
-                            e.getLocalizedMessage()));
+        ConnectivityManager connectivityManager =
+                ServiceProvider.getInstance().getAppContextService().getConnectivityManager();
+        if (connectivityManager == null) {
+            return ConnectionStatus.UNKNOWN;
         }
 
-        return DeviceInfoService.ConnectionStatus.UNKNOWN;
+        return NetworkUtils.isInternetAvailable(connectivityManager)
+                ? ConnectionStatus.CONNECTED
+                : ConnectionStatus.DISCONNECTED;
```

Comment:
I was going to suggest 
```
return (connectivityManager !=null  &&  NetworkUtils.isInternetAvailable(connectivityManager)) ?: ConnectionStatus.CONNECTED : ConnectionStatus.DISCONNECTED
```
but not sure if the existing clause is more clear to read. LGTM either way.

Reply:
`ConnectionStatus.UNKNOWN` makes more sense when connectivityManager is null.